xtable-core/src/main/java/org/apache/xtable/hudi/HudiConversionTarget.java

```diff
@@ -316,6 +323,53 @@ public String getTableFormat() {
     return TableFormat.HUDI;
   }
 
+  @Override
+  public Optional<String> getTargetCommitIdentifier(String sourceIdentifier) {
+    if (!metaClient.isPresent()) {
+      return Optional.empty();
+    }
+    return getTargetCommitIdentifier(sourceIdentifier, metaClient.get());
+  }
+
+  public Optional<String> getTargetCommitIdentifier(
+      String sourceIdentifier, HoodieTableMetaClient metaClient) {
+    long sourceIdentifierVal = Long.parseLong(sourceIdentifier);
+
+    HoodieTimeline completedTimeline = metaClient.getActiveTimeline().filterCompletedInstants();
```

Comment:
We can filter this to the commits timeline (avoids looking at cleaner or other commit types) with `getCommitsTimeline`